### PR TITLE
OverlayTiming.cc: mark overlay particles as overlay

### PR DESCRIPTION
**Pull request by Zehvogel**
The other Overlay processor does this correctly by using the merger for the merging, someone should probably refactor this one to also use that at some point?

I tested this change and it successfully adds the overlay flag to (only) the overlaid particles.

BEGINRELEASENOTES
- OverlayTiming.cc: mark overlay particles as overlay
- OverlayTiming.cc: mark overlay SimTrackerHits as overlay

ENDRELEASENOTES

### CODE REVIEW

**Comment by tmadlener** [comment]
What is *the other Overlay processor* in this context? Anyhow, this seems to be the right thing to do.

**Comment by Zehvogel** [comment]
By *other* I meant the one in `Overlay.cc`

**Comment by Zehvogel** [comment]
Oh but before you merge this let me also add setOverlay for the SimTrackerHits below this change

**Comment by Zehvogel** [comment]
Done, and it looks like isOverlay/setOverlay is not a thing for the SimCalorimeterHits so nothing to change there?

**Comment by tmadlener** [comment]
> it looks like isOverlay/setOverlay is not a thing for the SimCalorimeterHits

~Since each `SimTrackerHit` has a `OneToOneRelation` to an `MCParticle` this information is not duplicated.~ scratch that.

For `SimCalorimeterHit` multiple `MCParticle`s can in principle contribute, so it is not possible to decide on the Hit level whether it is overlay or not.

**Review by tmadlener — APPROVED** [review]

**Review by andresailer — APPROVED** [review]

**Comment by Zehvogel** [comment]
can we merge this then if we have so many approvals?

**Pull request merged**
